core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/ListCodec.java

```diff
@@ -115,11 +115,19 @@ public ByteBuffer encode(
       int size = input.getInt();
       List<ElementT> result = new ArrayList<>(size);
       for (int i = 0; i < size; i++) {
+        ElementT element;
         int elementSize = input.getInt();
-        ByteBuffer encodedElement = input.slice();
-        encodedElement.limit(elementSize);
-        input.position(input.position() + elementSize);
-        result.add(elementCodec.decode(encodedElement, protocolVersion));
+        // Allow null elements on the decode path, because Cassandra might return such collections
+        // for some computed values in the future -- e.g. SELECT ttl(some_collection)
+        if (elementSize < 0) {
+          element = null;
+        } else {
+          ByteBuffer encodedElement = input.slice();
+          encodedElement.limit(elementSize);
+          input.position(input.position() + elementSize);
```

Comment:
Nit: it would be a bit easier to understand this code if you invert these two lines:

```
element = elementCodec.decode(encodedElement, protocolVersion);
input.position(input.position() + elementSize);
```

(I initially thought that updating the position of `input` was required before the element can be decoded.)

Reply:
ok I'm doing it in a separate commit